packages/gatsby-plugin-manifest/src/gatsby-node.js

```diff
@@ -237,11 +235,9 @@ const makeManifest = async ({
 
       // if file names are being modified by cacheBusting icons must be generated after cache busting runs
       if (cacheMode !== `query`) {
```

Comment:
```suggestion
      if (cacheMode === `name`) {
```

If icons are to be generated when the file names are modified, then isn't it best to `generateIcon` when cacheMode is equal to `name` cuz that's when the filenames do get changed!

Reply:
Thanks for the suggestion but it's a change that's not relevant to this PR. Also, the code above is fine as it will catch anything not matching the above

Reply:
> Also, the code above is fine as it will catch anything not matching the above.

Makes sense!